masq/src/commands/financials_command.rs

```diff
@@ -438,77 +459,46 @@ impl FinancialsCommand {
         } else {
-            Self::convert_masq_from_gwei_and_dress_well(gwei)
+            //9,223,372,036
+            let gwei_as_i64 =
+                i64::try_from(gwei).expect("value bigger than i64::MAX is unrealistic");
+            Self::convert_masq_from_gwei_and_dress_well(gwei_as_i64)
         }
     }
 
-    fn abs_generic<T>(num: T, is_positive: bool) -> T
-    where
-        T: Display + PartialEq + PartialOrd<T> + From<u32> + Div<Output = T> + Bounded + CheckedNeg,
-        <T as Div<T>>::Output: PartialEq<T> + PartialOrd<T> + Display,
-    {
-        match (num == T::min_value(), is_positive) {
-            (true, false) => panic!("overflow at abs() of minimum signed value in an integer (however, MASQ amount {} is a nonsense in any case)", num),
-            (false, false) => num.checked_neg().expectv("negative number"),
-            (_, true) => num,
-        }
-    }
-
-    fn convert_masq_from_gwei_and_dress_well<T>(gwei: T) -> String
-    where
-        T: Display + PartialEq + PartialOrd<T> + From<u32> + Div<Output = T> + Bounded + CheckedNeg,
-        <T as Div<T>>::Output: PartialEq<T> + PartialOrd<T> + Display,
-    {
-        let (one_hundredth_masq_or_more, is_positive) = match Self::preformatting_math(gwei) {
-            Left(result) => return result,
-            Right(tuple) => tuple,
-        };
-
-        let sign_opt_with_all_digits = format!(
-            "{}{:0>3}",
-            if is_positive { "" } else { "-" },
-            Self::abs_generic(one_hundredth_masq_or_more, is_positive)
-        );
-        let full_len = sign_opt_with_all_digits.len();
-        let int_stem = &sign_opt_with_all_digits[..full_len - 2];
-        let fractional_part = &sign_opt_with_all_digits[full_len - 2..];
-        format!("{}.{}", int_stem.separate_with_commas(), fractional_part)
-    }
-
-    fn preformatting_math<T>(gwei: T) -> Either<String, (T, bool)>
-    where
-        T: Display + PartialEq + From<u32> + Div<Output = T> + PartialOrd<T>,
-        <T as Div<T>>::Output: PartialEq<T> + PartialOrd<T> + Display,
-    {
-        if gwei == T::from(0) {
-            return Left("0".to_string());
-        }
-        let is_positive = gwei >= T::from(0);
-        let one_hundredth_masq_or_more = gwei / T::from(10_000_000);
-        if one_hundredth_masq_or_more == T::from(0) {
-            return Left(
-                if is_positive {
+    fn convert_masq_from_gwei_and_dress_well(balance_gwei: i64) -> String {
+        //TODO neaten this ???maybe a const???
+        let balance_masq_int = (balance_gwei / 1_000_000_000).abs();
+        let balance_masq_frac = (balance_gwei % 1_000_000_000).abs();
+        let balance_masq_frac_trunc = balance_masq_frac / 10_000_000;
+        match (
+            (balance_masq_int == 0) && (balance_masq_frac_trunc == 0),
+            balance_gwei >= 0,
+        ) {
+            (true, bigger_then_zero) => {
+                let str_response = if bigger_then_zero {
```

Comment:
It's strange that you have an if inside a single match condition. Would it look better if this were two match conditions?

```
match (one_boolean, another_boolean) {
    (true, true) => "< 0.01".to_string(),
    (true, false) => "-0.01 < x < 0".to_string(),
    ...
```